packages/client/src/utils/utils.ts

```diff
@@ -118,3 +118,10 @@ export class MaxSizedSet<T> {
         this.delegate.delete(value)
     }
 }
+
+// A unique internal identifier to some list of primitive values. Useful
+// e.g. as a map key or a cache key.
+// - we could also support other primitives or optional arguments if needed
```

Comment:
This part a bit redundant imo. 
```suggestion
// - we could also support other primitives or optional arguments if needed
```